src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -21,6 +22,7 @@ namespace NuGet.Build.Tasks.Pack
     public class PackTaskLogic : IPackTaskLogic
     {
         private const string IdentityProperty = "Identity";
+        private const string NoWarn = nameof(NoWarn);
```

Comment:
```suggestion
        private const string NoWarnProperty = "NoWarn";
```

I would expect something called `NoWarn` to be the value of the property after an MSBuild evaluation, not a const string.